internal/example_web/main.go

```diff
@@ -27,30 +27,36 @@ func subProcess(ctx context.Context) {
 	span.Event("an event")
 }
 
+var pairs = []strc.HeadfieldPair{
+	{HeaderName: "X-Request-Id", FieldName: "request_id"},
+}
+
 func startServers(logger *slog.Logger) (*echo.Echo, *echo.Echo, *http.Server) {
-	middleware := strc.NewMiddleware(logger)
-	loggingMiddleware := strc.NewMiddlewareWithConfig(logger, strc.MiddlewareConfig{
+	m1 := strc.NewMiddleware(logger)
+	m2 := strc.NewMiddlewareWithConfig(logger, strc.MiddlewareConfig{
 		WithUserAgent:      true,
 		WithRequestBody:    true,
 		WithRequestHeader:  true,
 		WithResponseBody:   true,
 		WithResponseHeader: true,
 		Filters:            []strc.Filter{strc.IgnorePathPrefix("/metrics")},
 	})
+	m3 := strc.HeadfieldPairMiddleware(pairs)
```

Comment:
Example how to use this middleware - the incoming request ID header will be stored in context and then extracted and added as a log field to every single log message that is logged with the context.